### PR TITLE
Initial vault extension creation

### PR DESCRIPTION
**Pull request by AleksandrsKondratjevs**
Vault extension that allows to use any payment methods that has vault. Is instant purchanes active status defines if vault is enabled (default magento behavior). Main functionality is to add new tab in my-account with stored payments and allow to use them on checkout.

### CODE REVIEW

File: src/component/VaultStorage/VaultStorage.component.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    render() {
+        const { isLoading } = this.props;
+        return (
+            <>
+            <Loader isLoading={ isLoading } />
+            <div block="VaultStorage">
+                { this.renderContent() }
+            </div>
+            </>
```
Comment (yeegor) [inline src/component/VaultStorage/VaultStorage.component.js]: Doesn't ESLint shout at this indentation?

File: src/plugin/App.plugin.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import getStore from 'Store/index.js';
+
+import { VaultReducer } from '../store/Vault/Vault.reducer';
+
+/** @namespace VaultGraphql/Plugin/App/renderRedux */
+export const renderRedux = (args, callback, instance) => {
+    getStore().injectReducer('VaultReducer', VaultReducer);
+
+    return callback.apply(instance, args);
+};
```
Comment (yeegor) [inline src/plugin/App.plugin.js]: Please try to do it without importing anything from thee store. Maybe the store can be intercepted before rendering into the context providers? Sometimes direct imports from the store break the application having some irregular circumstances.

File: src/plugin/CheckoutBillingContainer.plugin.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+/** @namespace VaultGraphql/Plugin/CheckoutBillingContainer/mapStateToProps */
+export const mapStateToProps = (args, callback) => {
+    const [
+        {
+            VaultReducer: {
+                public_hash: selectedStoredPaymentMethod
+            }
+        }
+    ] = args;
```
Comment (yeegor) [inline src/plugin/CheckoutBillingContainer.plugin.js:22]: You can do that directly in the arguments section:
```js
export const mapStateToProps = ([{
    VaultReducer: { public_hash }
}], callback) => ...
```

File: src/plugin/CheckoutPaymentsContainer.plugin.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+/** @namespace VaultGraphql/Plugin/CheckoutPaymentsContainer/mapStateToProps */
+export const mapStateToProps = (args, callback) => {
+    const [
+        {
+            ConfigReducer: {
+                instance_purchase_active: isInstancePurchaseActive
+            }
+        }
+    ] = args;
```
Comment (yeegor) [inline src/plugin/CheckoutPaymentsContainer.plugin.js:20]: same here as above

File: src/store/Vault/Vault.dispatcher.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    static deletePaymentMethod(dispatch, options) {
+        const { public_hash } = options;
+        dispatch(updateVaultIsLoading(true));
+        fetchMutation(VaultQuery.getDeleteCardFromVaultMutation(
+            public_hash
+        )).then(
+            /** @namespace VaultGraphQl/Store/Vault/Dispatcher/getDeleteCardFromVaultMutationThen */
+            ({ deletePaymentToken: { storedPaymentMethods } }) => {
+                dispatch(updateStoredPaymentMethods(storedPaymentMethods));
+                dispatch(showNotification('success', __('Stored Payment Method was successfully removed!')));
```
Comment (yeegor) [inline src/store/Vault/Vault.dispatcher.js:41]: "is successfully removed" would be better. Or "has been removed successfully". "Was" tense implies action's incompleteness.

**Review by yeegor — CHANGES_REQUESTED** [review]
Minor changes required, but overall this is GOOOD!

File: src/store/Vault/Vault.dispatcher.js (hunk 5 above)
Comment (AleksandrsKondratjevs) [reply src/store/Vault/Vault.dispatcher.js:41]: Hey! I guess I just took this message from default magento (Luma theme). Should I still change it?

**Review by yeegor — APPROVED** [review]
Legit.

**Pull request merged**